lib/puppetserver/ca/sign_action.rb

```diff
@@ -0,0 +1,175 @@
+require 'puppetserver/ca/utils'
+require 'puppetserver/utils/http_utilities'
+require 'puppetserver/utils/file_utilities'
+require 'puppetserver/ca/puppet_config'
+require 'optparse'
+require 'openssl'
+require 'net/https'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class SignAction
+
+      include Puppetserver::Utils
+
+      SUMMARY = "Set up the CA's key, certs, and crls"
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca sign [--help|--version]
+  puppetserver ca sign [--config] --certname CERTNAME[,CERTNAME]
+  puppetserver ca sign  --all
+
+Description:
+Given a comma-separated list of valid certnames, instructs the CA to sign each cert.
+
+Options:
+      BANNER
+      BODY = JSON.dump({desired_state: 'signed'})
+
+      def self.parser(parsed = {})
+        OptionParser.new do |opts|
+          opts.banner = BANNER
+          opts.on('--certname x,y,z', Array, 'the name(s) of the cert(s) to be signed') do |cert|
+            parsed['certname'] = cert
+          end
+          opts.on('--config PUPPET.CONF', 'Custom path to Puppet Conf') do |conf|
```

Comment:
I would say "... to Puppet's config file". Puppet Conf is an event in San Francisco :P